src/Microsoft.Health.Dicom.Core/Features/ExtendedQueryTag/QueryTag.cs

```diff
@@ -45,6 +46,20 @@ public QueryTag(ExtendedQueryTagStoreEntry entry)
             ExtendedQueryTagStoreEntry = entry;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QueryTag"/> class.
+        /// </summary>
+        /// <remarks>Used for constructing from <see cref="WorkitemQueryTagStoreEntry"/> (to model sequences).</remarks>
+        /// <param name="entry">The WorkitemQueryTagStore entry.</param>
+        public QueryTag(WorkitemQueryTagStoreEntry entry)
```

Comment:
Consider separating out to different classes with a base abstract class

Reply:
Yeah agree, but as we discussed we will refactor to better approach later and keep this design for now.